agilent_infiniium_scope.py

```diff
@@ -31,446 +31,344 @@
-                ('xFirst',True),
-                ('xRef',True),
-                ('yStep',True),
-                ('yOrigin',True),
-                ('yRef',True),
-                ('coupling',True),
-                ('xRange',True),
-                ('xLeftDisplay',True),
-                ('yRange',True),
-                ('yCenterDisplay',True),
-                ('date',True),
-                ('time',True),
-                ('model',True),
-                ('acquisitionMode',True),
-                ('percentTimeBucketsComplete',True),
-                ('xUnits',True),
-                ('yUnits',True),
-                ('maxBW',True),
-                ('minBW',True)]
-    preambleDict = {}
-    for key,val in zip(preambleKeys,preambleVals):
-        if key[1]:
-            preambleDict[key[0]] = val
-    def unitType(num):
-        if num=='1':
+    preamble_vals = preamble.split(',')
+
+    for (key, included), val in zip(PREAMBLE_KEYS, preamble_vals):
+        if included:
+            preamble_dict[key] = val
+
+    def unit_type(num):
+        if num == '1':
             return 'V'
-        elif num=='2':
+        elif num == '2':
             return 'ns'
         else:
             raise Exception('Units not time or voltage')
-    preambleDict['xUnit'] = unitType(preambleDict['xUnits'])
-    preambleDict['yUnit'] = unitType(preambleDict['yUnits'])
-    return (preambleDict)
+    preamble_dict['xUnit'] = unit_type(preamble_dict['xUnits'])
+    preamble_dict['yUnit'] = unit_type(preamble_dict['yUnits'])
+    return preamble_dict
+
 
-def _parseBinaryData(data, wordLength):
+def _parseBinaryData(data, word_length):
     """Parse binary data packed as string of RIBinary
+
+    Data format discussed here:
+    http://www.keysight.com/upload/cmc_upload/All/9000_series_prog_ref.pdf?&cc=US&lc=eng#page=1091
     """
-    formatChars = {'1':'b','2':'h', '4':'f'}
-    formatChar = formatChars[str(wordLength)]
-
-    #Get rid of header crap
-    #unpack binary data
-    if wordLength == 1:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = numpy.array(unpack(formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 2:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))
-    elif wordLength == 4:
-        lenHeader = int(data[1])
-        dat = data[(2+lenHeader):]
-        dat = dat[-calcsize('>' + formatChar*(len(dat)/wordLength)):]
-        dat = numpy.array(unpack('>' + formatChar*(len(dat)/wordLength),dat))      
-    return dat
+    format_chars = {1: 'b', 2: 'h', 4: 'f'}
+    format_char = format_chars[word_length]
+
+    # Get rid of header
+    # unpack binary data
+    if data[0] != '#':
+        raise Exception("Invalid wave data. Expected '#' "
+                        "at start. Expected '#' but got {}".format(data[0]))
```

Comment:
Just `"at start but got {}".format(data[0]))`. No need to repeat expected value.